### PR TITLE
fix: changes requested in #2

### PR DESCRIPTION
**Pull request by merrcury**
Changes as per #2

### CODE REVIEW

**Review by samcaspus — APPROVED** [review]
looks good to me CC @yakkomajuri

File: index.js [hunk 1]
```diff
@@ -27,73 +24,31 @@ export async function setupPlugin({ config, global, cache }) {
...
-      user: {
-        user_fields: { [eventName]: `${sent_at}` },
-      },
-    });
-
-    const retryResponse = await fetchWithRetry(
-      `${global.fetchUserUrl}/${userId}`,
-      updatedHeaders,
+export const jobs = {
+  pushUserDataToZendesk: async (request, meta) => {
```
Comment (yakkomajuri) [inline index.js]: Let's have this job do even more. The basic event validation should be done on `onEvent` but from the storage call onwards let's do things here.

File: index.js [hunk 2]
```diff
@@ -163,37 +118,40 @@ async function fetchAllTickets(global, storage, cache) {
...
-      await pushUserDataToZendesk(
-        email,
-        event.event,
-        event["sent_at"],
-        global,
-        storage
-      );
+      const userId = await storage.get(email);
+      if (userId) {
+        const url = await global.fetchUserUrl;
```
Comment (yakkomajuri) [inline index.js]: this doesn't need an await

**Review by yakkomajuri — CHANGES_REQUESTED** [review]
partial review, just jumped on a call

File: index.js (hunk 1 above)
Comment (yakkomajuri) [inline index.js]:
```suggestion
  pushUserDataToZendesk: async (request, { storage, cache, global }) => {
```

Comment (samcaspus) [reply index.js]: Thanks @yakkomajuri  will make the changes and update soon

File: index.js [hunk 3]
```diff
@@ -27,73 +24,43 @@ export async function setupPlugin({ config, global, cache }) {
...
-      updatedHeaders,
-      "PUT"
-    );
-    await retryResponse.json();
-  }
-}
+    if (userId) {
+      const url = global.fetchUserUrl;
 
+      global.defaultHeaders.body = JSON.stringify({
```
Comment (yakkomajuri) [inline index.js]: nit: this is not a heder, it's the body of the request

File: index.js [hunk 4]
```diff
@@ -27,73 +24,43 @@ export async function setupPlugin({ config, global, cache }) {
...
-      options,
-      (method = method),
-      (isRetry = true)
-    );
-    return res;
-  }
-}
+export const jobs = {
+  pushUserDataToZendesk: async (request, { storage, global, cache }) => {
+    const userId = await storage.get(request.email);
```
Comment (yakkomajuri) [inline index.js]:
```suggestion
    const userId = await storage.get(request.email, null);
```

**Review by yakkomajuri — CHANGES_REQUESTED** [review]
looking much better, thanks. a few comments

**Review by yakkomajuri — APPROVED** [review]

**Pull request merged**